device.go

```diff
@@ -166,11 +177,85 @@ func newBlockDevice(devInfo DeviceInfo) *BlockDevice {
 	}
 }
 
-func (device *BlockDevice) attach(h hypervisor) error {
+func makeBlockDevIDForHypervisor(deviceID string) string {
+	return fmt.Sprintf("drive-%s", deviceID)
+}
+
+func (device *BlockDevice) attach(h hypervisor, c *Container) error {
```

Comment:
In order not to forget to decrement in case of error, you should use `defer` here:
```go
func (device *BlockDevice) attach(h hypervisor, c *Container) (err error) {

        ...

        index, err := c.pod.getAndSetPodBlockIndex()
        if err != nil {
                return err
        }
        defer func() {
                if err != nil {
                         c.pod.decrementPodBlockIndex()
                }
        }()

        ...

}
```

Reply:
I am using the util function generateRandomBytes with parameter 8 to get the id/node-name passed to qemu, so the length of the encoded string will be twice that + len("drive-").
I am not using uuid anymore, which similarly read from the rand device, but generated a string of more than 32 characters.

Reply:
Ok I understand that you're solving the problem here, but I would expect a check to ensure the final string won't be more than 31 characters. If someday someone is changing `fmt.Sprintf("drive-%s", deviceID)` to `fmt.Sprintf("super-long-drive-string-%s", deviceID)`, we are gonna hit the 31bytes issue. Actually, I think this should be enforced inside the Qemu implementation since this is Qemu specific. We need a function to truncate IDs so that either if you hotplug or hotunplug, Qemu will always provide a consistent string of less than 32 bytes.

Also, adding a comment about that should help anyone reading the code.